File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-060430-47161/output/project/src/main/java/surveycrawler/Silt.java
package surveycrawler;

public class Silt extends GroundType {

    // moving

    @Override
    public void moveTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.slideOnSiltTowards(aDirection);
    }

    // facing

    @Override
    public void assertCanTurn(SurveyCrawler aCrawler) {
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-212215-18237/analysis/own-tests-project/src/surveycrawler/silt.py
import random

from surveycrawler.ground_type import GroundType


class Silt(GroundType):

    # instance creation

    @classmethod
    def with_random(cls, a_random):
        return cls(a_random)

    # sliding

    @classmethod
    def sliding_limit(cls):
        return 10

    # initialization

    def __init__(self, a_random=None):
        self._random = a_random if a_random is not None else random.Random()

    # moving

    def move_crawler_to(self, a_crawler, a_position, a_direction):
        a_slide = a_direction.times(self._random.randrange(self.sliding_limit()))
        a_crawler.slide_to(a_position.plus(a_slide))

    # facing

    def turn_crawler_with(self, a_crawler, a_turn):
        a_turn()
